services/movie_session.py

```diff
@@ -0,0 +1,69 @@
+from typing import Optional
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.db.models import QuerySet
+
+from db.models import MovieSession, Movie, CinemaHall
+from datetime import datetime
+
+
+def create_movie_session(movie_show_time: datetime,
+                         movie_id: int,
+                         cinema_hall_id: int) -> MovieSession:
+    try:
+        movie = Movie.objects.get(id=movie_id)
+        cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
+    except ObjectDoesNotExist:
+        raise ValueError("Provided movie or cinema hall ID does not exist.")
+
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie=movie,
+        cinema_hall=cinema_hall
+    )
+
+
+def get_movies_sessions(session_date: str = None) -> QuerySet[MovieSession]:
+    if session_date:
+        try:
+            date_obj = datetime.strptime(session_date, "%Y-%m-%d").date()
+        except ValueError:
+            raise ValueError("Invalid date format. Expected 'year-month-day'")
```

Comment:
The error message in the `ValueError` could be more informative by specifying the expected format explicitly, such as 'YYYY-MM-DD'.